api/apps/geoprocessing/src/modules/planning-units-grid/planning-units-grid.processor.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Job, Worker } from 'bullmq';
+import { plainToClass } from 'class-transformer';
+import { validateSync } from 'class-validator';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+
+import { JobInput, JobOutput, queueName } from '@marxan/planning-units-grid';
+import { ShapefileService } from '@marxan/shapefile-converter';
+import {
+  WorkerBuilder,
+  WorkerProcessor,
+} from '@marxan-geoprocessing/modules/worker';
+
+import { ShapeType } from '@marxan-jobs/planning-unit-geometry';
+import { GridGeoJsonValidator } from './grid-geojson-validator';
+import { isLeft } from 'fp-ts/Either';
+
+@Injectable()
+export class PlanningUnitsGridProcessor
+  implements WorkerProcessor<JobInput, JobOutput> {
+  private readonly worker: Worker;
+  private readonly logger = new Logger(this.constructor.name);
+
+  constructor(
+    @InjectEntityManager() private readonly entityManager: EntityManager,
+    private readonly shapefileConverter: ShapefileService,
+    private readonly gridGeoJsonValidator: GridGeoJsonValidator,
+    workerBuilder: WorkerBuilder,
+  ) {
+    this.worker = workerBuilder.build(queueName, this);
+  }
+
+  async process({
+    data: { projectId, shapefile },
+  }: Job<JobInput, JobOutput>): Promise<JobOutput> {
+    const { data: geoJson } = await this.shapefileConverter.transformToGeoJson(
+      shapefile,
+    );
+
+    const result = this.gridGeoJsonValidator.validate(geoJson);
+    if (isLeft(result)) {
+      throw new Error(result.left.toString());
+    }
+    const puGeometriesIds: { id: string }[] = await this.entityManager.query(
+      `
+        INSERT INTO "planning_units_geom"("the_geom", "type", "project_id")
+        SELECT ST_SetSRID(
+                 ST_GeomFromGeoJSON(features ->> 'geometry'),
+                 4326)::geometry,
+               $2,
+               $3
+        FROM (
+               SELECT json_array_elements($1::json -> 'features') AS features
+             ) AS f
+        ON CONFLICT ON CONSTRAINT planning_units_geom_the_geom_type_key DO UPDATE SET type = 'irregular'
```

Comment:
` type`  should be inputted from the FE

Reply:
Well, it's a pity we haven't mentioned this in requirements.

Are we sure this makes sense? Allowing user to provide whatever shape they want and marking the as "hex" (which isn't correct, right?) may lead system to produce unexpected results.

Reply:
The user will upload their custom planning grid, which could be a regular (square, hexagon, triangles...) or an irregular one( whatever continuos group of geometries that can represent basins, parcels...). 
For those regulars, the main difference with the ones we create is related to the grid upper left corner origin. If we want to be agnostic to the type we can make use of the same categorization in the enum planning_unit_grid_shape and mark them as from_shapefile. 
But marking it as irregular is just wrong. What do you think?

Reply:
@aagm I think following https://github.com/Vizzuality/marxan-cloud/pull/526#discussion_r705954882 at this very moment would be fine. (in future we can make the validation/checks also instead asking for a type, we will need to verify it anyway so we can also deduce it, right?) Nevertheless, as long as it isn't that much concern for this moment, `from_shapefile` sounds fair enough for current state.

Reply:
updated in https://github.com/Vizzuality/marxan-cloud/pull/525/commits/05039600456c4312c73e78d2b4e412948b166c9b